hail/python/hail/expr/functions.py

```diff
@@ -6984,6 +6984,43 @@ def shuffle(a, seed: Optional[builtins.int] = None) -> ArrayExpression:
     return sorted(a, key=lambda _: hl.rand_unif(0.0, 1.0))
 
 
+def __validate_and_coerce_endpoint(point, key_typ):
+    """query validation for the points or endpoints of the query in query_table"""
+    len = builtins.len
+    key_names = list(key_typ)
+    if point.dtype == key_typ[0]:
+        point = hl.struct(**{key_names[0]: point})
+    ts = point.dtype
+    if not isinstance(ts, tstruct):
+        raise ValueError(
+            f'key mismatch: cannot use query point type {point.dtype} to query a table with key of '
+            f'({", ".join(builtins.str(x) for x in key_typ.values())}) '
+        )
+
+    if not ts:
```

Comment:
I dislike relying on truthiness. Could you make this an explicit test like `len(ts) == 0`?

Reply:
```suggestion
    if len(ts) == 0:
```
Okay I'll change it.

Just so you know, in the absence of a `__bool__` method on a collection class, `__len__` is called and compared to 0. So, these are in fact, identical.

Reply:
Sure, I just don't think readers should be forced to know that (and remember that `tstruct` is a collection), or wonder if we've defined some weird `__bool__` method. Thanks for making the change!